### PR TITLE
Refreshes the UI instantly as a filter is applied.

### PR DESCRIPTION
**Pull request by shreyas-satish**

### CODE REVIEW

File: hasjob/templates/layout.html [hunk 1]
```diff
@@ -84,19 +84,19 @@
                 <ul class="dropdown-menu pay-filter-dropdown" role="menu" aria-labelledby="job-filters-pay">
                   <li class="clearfix">
                     <div class="currency-checkbox">
-                      <input type="radio" id="job-filters-na" name="currency" value="NA"></input>
+                      <input type="radio" class='js-handle-filter-change' id="job-filters-na" name="currency" value="NA"></input>
```
Comment (jace) [inline hasjob/templates/layout.html]: Shouldn't we standardise on double quotes for all attribute values?

**Comment by jace** [comment]
Instant refresh loses the `archive` flag and possibly others.

**Comment by jace** [comment]
Managed to break it by hitting Back from a jobpost.

screen shot 2015-11-18 at 2 49 07 am

**Comment by jace** [comment]
`window.Hasjob` become undefined in this case, but I'm unable to reproduce it again. I guess something to do with `pushState` and how `app.js` is referred to in different templates?

**Comment by jace** [comment]
The Back button issue was caused by the way Chromium handles cache. Switching to `POST` for loading fragments has solved this. https://code.google.com/p/chromium/issues/detail?id=94369

**Comment by shreyas-satish** [comment]
It's also worth noting that Safari fires the [popstate](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onpopstate) event on page load. This is a [bug](https://code.google.com/p/chromium/issues/detail?id=63040) in Safari that used to be present in Chromium too, but has now been fixed. I'm currently testing potential work-arounds. 

[History.js](https://github.com/browserstate/history.js) is an option, but that comes with the following quirk - it fires the `onstatechange` event _both_ when a page is arrived at through a back/forward button and through `pushState`. This behaviour makes it confusing to decide when to initiate a reload.

**Pull request merged**